httpcore/_async/connection_pool.py

```diff
@@ -124,7 +124,10 @@ async def request(
         stream: AsyncByteStream = None,
         timeout: TimeoutDict = None,
     ) -> Tuple[bytes, int, bytes, Headers, AsyncByteStream]:
-        assert url[0] in (b"http", b"https")
+        if url[0] not in (b"http", b"https"):
+            scheme = url[0].decode("latin-1")
+            raise UnsupportedProtocol("Unsupported URL protocol '{scheme}'")
```

Comment:
Missing `f` for this f-string :-) Also using `!r` to automatically show the `'` from the string repr…

```suggestion
            raise UnsupportedProtocol(f"Unsupported URL protocol {scheme!r}")
```